### ISSUE TITLE
Enabling 2FA on Gmail Breaks Currently Running Instance

### ISSUE THREAD

**Issue by ghost** [Foundry376/Mailspring#299]
Running latest rev of mailspring on Ubuntu 16.04
Enabled 2FA on my gmail account
Mailspring goes autistic -- says it's syncing but never completes.
(Let the sync process run overnight -- no change - still syncing...)
Re-enabled Mailspring's authentication....no change...
Quit Mailspring...
Re-enabled Mailsprings authentication (on gmail)... no change except mailspring no longer starts-up the gui window
Check process-status table... lots of debris....
Re-quit Mailspring....
Wait...
Still processes running...(see below)...
Manually kill everything...
restart...
Mailspring comes back, authenticated and giving access back to inbox after several mins of processing...

tl;dr:  quit mailspring completely before enabling 2FA, then re-start and re-auth.

`
# ps -ef | grep -i mail
mshallop  9127  3221  1 Oct30 ?        01:37:20 /usr/share/mailspring/mailspring
mshallop  9130  9127  0 Oct30 ?        00:00:00 /usr/share/mailspring/mailspring --type=zygote --no-sandbox
mshallop  9166  9127  1 Oct30 ?        01:33:05 /usr/share/mailspring/mailspring --type=gpu-process --no-sandbox --supports-dual-gpus=false --gpu-driver-bug-workarounds=1,7,23,61,74 --disable-gl-extensions=GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent --gpu-vendor-id=0x8086 --gpu-device-id=0x1916 --gpu-driver-vendor --gpu-driver-version --gpu-driver-date --service-request-channel-token=F0D5A05C47FFF709C3624140688D1C36
mshallop  9193  9130  1 Oct30 ?        01:30:26 Mailspring default
mshallop 10471  9130  0 Oct31 ?        00:00:03 Mailspring emptyWindow
root     21740  5630  0 07:35 pts/1    00:00:00 grep --color=auto -i mail
mshallop 28236  9130  0 Oct31 ?        00:00:02 /usr/share/mailspring/mailspring --type=renderer --js-flags=--harmony --no-sandbox --primordial-pipe-token=ED09BEC8799EDFCE42A2731A3D74A200 --lang=en-US --app-path=/usr/share/mailspring/resources/app.asar --node-integration=true --webview-tag=true --no-sandbox --enable-pinch --num-raster-threads=2 --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,3553;0,11,3553;0,12,3553;0,13,3553;0,14,3553;0,15,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,3553;1,11,3553;1,12,3553;1,13,3553;1,14,3553;1,15,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,3553;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,3553;2,11,3553;2,12,3553;2,13,3553;2,14,3553;2,15,3553;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,3553;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,3553;3,13,3553;3,14,3553;3,15,3553;4,0,3553;4,1,3553;4,2,3553;4,3,3553;4,4,3553;4,5,3553;4,6,3553;4,7,3553;4,8,3553;4,9,3553;4,10,3553;4,11,3553;4,12,3553;4,13,3553;4,14,3553;4,15,3553 --disable-accelerated-video-decode --service-request-channel-token=ED09BEC8799EDFCE42A2731A3D74A200 --renderer-client-id=34 --shared-files=v8_natives_data:100,v8_snapshot_data:101
mshallop 31104  9193  0 Nov02 ?        00:00:00 /bin/bash /usr/share/mailspring/resources/app.asar.unpacked/mailsync --mode sync
mshallop 31113 31104  0 Nov02 ?        00:00:57 /usr/share/mailspring/resources/app.asar.unpacked/mailsync.bin --mode sync
`

**Comment by bengotow** [comment]
Hey! I think this may be several different things combined into one. Mailspring should not lose access when you enable 2FA unless you explicitly revoke it. It also doesn't immediately lose access even when you /do/ revoke it's token,  because the existing connections to the IMAP server can stay open for hours even when new ones would fail. (Which honestly is a bit of a security bug on their side I think.) 

Any chance Mailspring was correctly syncing, and just indexing messages all night? That's not uncommon for huge mailboxes that receive a large amount of email. We really need to get a progress bar in there.

Orphaned processes should appear when you force-quite mailspring should disappear within two minutes - there's actually a timeout in them so that you can do things like detach them and bind them to a debugger so I couldn't make them exit instantly.

**Comment by mshallop** [comment]
Thank you for the reply @bengotow ...

For clarification, I didn't have 2FA enabled on Mailspring -- I've been running MS for a couple versions, then enabled 2FA on my gmail account - a progress bar would be awesome, indicating sync progress... my mailbox isn't that large but I know it still takes several minutes to sync completely.

**Comment by CodeMouse92** [comment]
It appears that there has been no activity on this issue for some time, so we will close it. **If the problem occurs again, please create a new thread in the appropriate category on our [Discourse](https://community.getmailspring.com).**

We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

We hope to see you there soon!

-The Mailspring Team
